static/js/stream_data.js

```diff
@@ -301,6 +301,14 @@ exports.get_announcement_only = function (stream_name) {
     return sub.is_announcement_only;
 };
 
+exports.get_first_message_id = function (stream_name) {
+    var sub = exports.get_sub(stream_name);
+    if (sub === undefined) {
+        return false;
```

Comment:
@timabbott I updated PR to include all changes you proposed, but I don't understand the logic behind this PR.
If `all_msg_list.first_message_id <= sub.first_message_id` does it mean that every time we have first message in narrowed view then we display `more topics` and otherwise not? I think it is still possible to have more topics to display in the left sidebar

Reply:
Under what circumstance do you think it'd be possible?

There might need to be a few more checks here on `all_msg_list` being up-to-date that are a little subtle, but the point of this feature is to be able to avoid showing "more topics" when we know that all topics in the stream's history are already displayed.  

And since `all_msg_list` has a local copy of the history of all streams in a range of message IDs from `all_msg_list.first_message_id` through current, this value should let us do this check.